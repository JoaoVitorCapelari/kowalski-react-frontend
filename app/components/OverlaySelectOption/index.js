/**
*
* OverlaySelectOption
*
*/

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Option = styled.a`
  display: block;
`;

const SelectedOption = styled.a`
  display: block;
  color: blue;
`;

function OverlaySelectOption({ onOptionSelect, selectedOption, children, value }) {
  const OptionElement = selectedOption === value || children
    ? SelectedOption
    : Option;
  const finalValue = value || children;

  return (
    <OptionElement onClick={() => onOptionSelect(finalValue)}>
      { children }
    </OptionElement>
  );
}

OverlaySelectOption.propTypes = {
  onOptionSelect: PropTypes.func,
  selectedOption: PropTypes.any,
  children: PropTypes.any,
  value: PropTypes.any,
};

export default OverlaySelectOption;
