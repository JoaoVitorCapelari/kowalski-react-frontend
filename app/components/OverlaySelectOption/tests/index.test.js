import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import OverlaySelectOption from '../index';

const mountComponent = (selectedOption) => mount(
  <OverlaySelectOption {...{ selectedOption }}>
    Some text
  </OverlaySelectOption>
);

describe('<OverlaySelectOption />', () => {
  let renderedComponent;
  describe('when the option is selected', () => {
    beforeEach(() => {
      renderedComponent = mountComponent(true);
    });

    it('Renders one anchor tag blue colored', () => {
      expect(renderedComponent.html()).toMatchSnapshot();
      const anchors = renderedComponent.find('a');
      expect(anchors.length).toEqual(1);
      expect(anchors.first()).toHaveStyleRule('color', 'blue');
    });
  });

  describe('when the option is not selected', () => {
    beforeEach(() => {
      renderedComponent = mountComponent(false);
    });

    it('Renders one anchor not blue colored', () => {
      expect(renderedComponent.html()).toMatchSnapshot();
      const anchors = renderedComponent.find('a');
      expect(anchors.length).toEqual(1);
      expect(anchors.first()).not.toHaveStyleRule('color', 'blue');
    });
  });
});
