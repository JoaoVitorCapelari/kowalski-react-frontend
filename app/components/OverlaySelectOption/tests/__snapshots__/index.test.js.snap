// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverlaySelectOption /> when the option is not selected Renders one anchor not blue colored 1`] = `"<a class=\\"OverlaySelectOption__Option-douNpL tMoZj\\">Some text</a>"`;

exports[`<OverlaySelectOption /> when the option is selected Renders one anchor tag blue colored 1`] = `"<a class=\\"OverlaySelectOption__SelectedOption-lkCHwo gNWKrc\\">Some text</a>"`;
